src/rdb.c

```diff
@@ -3374,6 +3376,31 @@ static void backgroundSaveDoneHandlerSocket(int exitcode, int bysignal) {
 /* When a background RDB saving/transfer terminates, call the right handler. */
 void backgroundSaveDoneHandler(int exitcode, int bysignal) {
     int type = server.rdb_child_type;
+
+    if (bysignal == SIGUSR1) {
+        /* The background process was canceled by the user */
+        server.last_bgsave_failure_reason = SAVE_FAILURE_CANCELED;
+    } else if (bysignal) {
+        /* Assume all other signals are caused by OOM */
+        server.last_bgsave_failure_reason = SAVE_FAILURE_OOM;
```

Comment:
why do we make this assumption?
same for the disk and socket assumptions below..

Reply:
I guess we can limit it to SIGKILL and SIGTERM

Regarding the SAVE_FAILURE_DISK and SAVE_FAILURE_SOCKET. maybe the names are misleading.
The purpose was to indicate that there was an error with bgsave sending to socket/disk. maybe we can narrow down to just a simple BACKGROUND_SAVE_ERROR.

Reply:
let's first finish the discussion on the higher level [issue](https://github.com/redis/redis/pull/11514#pullrequestreview-1187869341) (in case you missed that comment)